src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/Null.ReadTests.cs

```diff
@@ -222,18 +222,39 @@ public static void DeserializeDictionaryWithNullValues()
             }
         }
 
-        [Fact]
-        public static void InvalidRootOnRead()
+        [Theory]
+        [InlineData(typeof(UIntPtr))]
+        [InlineData(typeof(IntPtr))]
+        [InlineData(typeof(Type))]
+        [InlineData(typeof(int[,]))]
```

Comment:
I think we should open the issue and try to support the multidimensional dictionaries

Reply:
I checked the backlog and there doesn't appear to be a request for that so far. I think we should leave it up to the users to pick a JSON representation that suites their requirements (for example, a jagged array would work for most cases, but it might not be the most compact representation when needing to marshal matrices).